src/openrct2-ui/windows/TileInspector.cpp

```diff
@@ -999,9 +998,13 @@ static void window_tile_inspector_mousedown(rct_window* w, rct_widgetindex widge
         return;
     }
 
-    switch (w->tileInspectorPage)
+    const TileElement* tileElement = window_tile_inspector_get_selected_element(w);
+    if (tileElement == nullptr)
+        return;
+
+    switch (tileElement->GetType())
```

Comment:
Shouldn't we have a validate function that confirms the type matches the page. You can't check widget addresses across pages so any miss match will cause very unexpected behaviour.

Reply:
Probably not a bad idea. It _should_ be impossible for the selected elements's type and current page to not match, but an assert and exit should at least avoid crashes.

Reply:
It would happen if your on a server and two people edit the same tile

Reply:
> assert and exit should at least avoid crashes.

That is still a crash. So better to just return and not show anything.

Reply:
>It would happen if your on a server and two people edit the same tile

If two players edit the same tile, the local selection index is updated accordingly, so it should be impossible still. That's why an assert may be preferable here, as that way it'll be found out when it gets broken instead of failing silently.

I may be misunderstanding something, but asserts do not crash the game, right? They just show a message that can be clicked away and the game continues its execution.